api/preload.js

```diff
@@ -0,0 +1,45 @@
+import fs from 'socket:fs/promises'
+import Path from 'socket:path'
+import process from 'socket:process'
+
+const preloadJavascript = `
+/// PRELOAD_JS_PLACEHOLDER ///
+`
+export const getPreloadJavascript = () => {
+  return preloadJavascript
+}
+
+const preloadScriptModule = `<script type="module">\n${preloadJavascript}\n</script>\n`
+
+const recursePath = async (path, fileFunc, data) => {
```

Comment:
Please use `function`, instead of arrow functions. It is easier to identity a function instead of a variable.

Can you make this function a little easier to read to?

```js
/**
 * @param {string} directory - Directory tree to walk down
 * @param {(string, any?): any} callback - Callback to invoke for each entry in directory tree
 * @return {Promise}
 */
async function walk (directory, callback) {
  const entries = await fs.readdir(directory, { withFileTypes: true })
  for (const entry of entries) {
    const filename = path.join(directory, entry.name)
    if (entry.isDirectory()) {
      await walk(filename, callback)
    } else {
      await callback(filename)
    }
  }
}
```